examples/gno.land/p/gov/dao/proposal.gno

```diff
@@ -0,0 +1,47 @@
+package dao
+
+import "std"
+
+type Status string
+
+// ACTIVE -> ACCEPTED -> EXECUTED
+// ACTIVE -> NOT ACCEPTED
+var (
+	Active      Status = "active"       // proposal is still active
+	Accepted    Status = "accepted"     // proposal gathered quorum
+	NotAccepted Status = "not accepted" // proposal failed to gather quorum
+	Executed    Status = "executed"     // proposal is executed
+)
+
+func (s Status) String() string {
+	return string(s)
+}
+
+// PropStore defines the proposal storage abstraction
+type PropStore interface {
+	// GetProposals returns the given paginated proposals
+	GetProposals(offset, count uint64) []Proposal
```

Comment:
maybe add a filter like active=true

or a dedicated helper

Reply:
same here, we should not return a slice with all the proposals, but helper to get a specific one by ID, to know the size, and eventually to paginate, but by default, the API should discourage listing more than the last 10 ones or getting a specific one

it's a bad gas optimization and a way to hit memory limit to let the current listing for such potentially-infinite objects.

Reply:
I agree, we shouldn't expose this kind of infinite logic, especially on an open API like this (anyone can get the PropStore)

I've limited it to a sane default (`10`):

54c555d